lib/mongo/crypt/status.rb

```diff
@@ -84,6 +84,23 @@ def ref
         @status
       end
 
+      # Raises a Mongo::Error:CryptError corresponding to the
+      # information stored in this status
+      #
+      # Does nothing if self.ok? is true
+      def raise_crypt_error
+        return if ok?
+
+        error = case label
+        when :error_kms
+          Error::CryptKmsError.new(code, message)
+        when :error_client
+          Error::CryptClientError.new(code, message)
+        end
+
+        raise error
```

Comment:
Symbols cannot be raised in Ruby, this will need to be revisited in the future to perhaps raise a base class of CryptError.